hs_access_control/models/user.py

```diff
@@ -318,6 +327,40 @@ def owned_groups(self):
     # access checks for groups
     #################################
 
+    # There is a duality between listing and access for groups.
+    # The group interface requires owns_group, views_group, etc to concern
+    # direct privilege. By contrast, the control functions can_*_group_*
+    # must include indirect privilege. Thus the listing functions _all_*_groups
+    # include indirect privilege
+
+    @property
+    def __all_view_groups(self):
+        """
+        Get a list of active groups accessible to self for view.
+
+        Inactive groups will be included only if self owns those groups.
+
+        :return: QuerySet evaluating to held groups.
+
+        This returns all groups that are viewable, including owned groups and groups
+        in communities to which the user has view access.
+        """
+        if not self.user.is_active:
+            raise PermissionDenied("Requesting user is not active")
+
+        return Group.objects.filter(
+                # owners can see inactive groups they own
+                Q(g2ugp__user=self.user,
+                  g2ugp__privilege=PrivilegeCodes.OWNER) |
+                # everyone else can see only active groups they are in
+                Q(gaccess__active=True,
+                  g2ugp__user=self.user) |
+                # if user has access to a group in a community, grant access to whole community
+                Q(gaccess__active=True,
```

Comment:
This filter condition `gaccess_active=True` is probably not needed here if it is looing for groups that are part of the community.

Reply:
Let's not tempt fate. Do you actually think that, e.g., Christina, will put a group into a community and then not disable the group?